src/app/courses/courses-list/course-item/course-item.component.ts

```diff
@@ -0,0 +1,29 @@
+import { DatePipe } from '@angular/common';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-course-item',
+  templateUrl: './course-item.component.html',
+  styleUrls: ['./course-item.component.css']
+})
+export class CourseItemComponent implements OnInit {
+  @Input() title: string = '';
+  @Input() description: string = '';
+  @Input() duration: number | undefined;
+  @Input() creationDate:  Date | undefined;
+  @Input() id: number | undefined;
+  @Output() deletedItemId = new EventEmitter<number>();
+
+  constructor(private datePipe: DatePipe) {}
```

Comment:
The idea is not bad, but Pipes are not injected in the constructor, they are imported in the module and then can be used in its scope

Reply:
remove from the contructor